AK/StringUtils.h

```diff
@@ -38,6 +39,13 @@ enum class TrimWhitespace {
     No,
 };
 
+enum class OnSplit : unsigned {
+    Nothing = 0,
+    KeepSeparator = 1,
+    KeepEmpty = 2,
```

Comment:
How come a boolean argument turned into three states?

Reply:
Because I added KeepSeparator, which is used by #15675, and it made sense to be an argument of split* functions

Reply:
Do that in a separate commit then